builtin/mainmenu/dlg_server_list_mods.lua

```diff
@@ -0,0 +1,96 @@
+-- Luanti
+-- Copyright (C) 2024 cx384
+-- SPDX-License-Identifier: LGPL-2.1-or-later
+
+local function get_formspec(dialogdata)
+	local TOUCH_GUI = core.settings:get_bool("touch_gui")
+	local server = dialogdata.server
+	local filter_prefix = dialogdata.filter_prefix
+
+	-- A wrongly behaving server may send ill formed mod names
+	local mods = {}
+	table.sort(server.mods)
+
+	if filter_prefix then
+		for _, mod in ipairs(server.mods) do
+			if mod:sub(0, #filter_prefix) == filter_prefix then
+				table.insert(mods, core.formspec_escape(mod))
+			end
+		end
+	else
+		local last_prefix
+		for i, mod in ipairs(server.mods) do
+			local prefix = mod:match("([^_]*_)")
```

Comment:
An idea here would be to use `([^_]*)_` as the match string. This could then be combined with a `mod == prefix` check to group (e.g.) `mesecons` and `mesecons_*` mods.

Reply:
Ok, now singletons are grouped together with the prefixes. e.g. `bucked` and `bucked_wooden` are grouped.